### DISCUSSION TITLE
SCAN_INTERVAL

### DISCUSSION THREAD

**Discussion by leeyuentuen** [leeyuentuen/alfen_wallbox#114] [General]
@ikruglov,
about the SCAN_INTERVAL, at the moment it has been changed to 30sec.
don't know if I'm the only one, but I have some automation to change the limit Amps between 6-16A depending on the usage.
Because here in Belgium we have 'capacity price' it means the highest Amps that you have to use the latest 15min, it will become the price calculation for the rest of the month. so I'm trying to limit my total usage below 3.5KW,
So the reason why I'm trying to put the scan interval as low as possible is to have the possibility to switch to lower amps and even stop the charging if I'm using too much power

**Comment by ikruglov** [comment]
@leeyuentuen.

I think the difference for automation between 5s and 30s interval is not significant.
However, I do not fully understand the situation with your price calculation. This is where it may play a role. I'm not an expert.
Why you can't simply limit charging at, for instance, fixed lower Amps value?

Maybe we should make this parameter configurable. Or, if I'm the only one I can just do local changes.

If you insist on keeping 5s interval, we need to do two adjustments:
1. keepalive_timeout should be `max(2 * interval, 15)`, so you don't lower timeout below the default value
2. adjust TIMEOUT. Having 90s timeout with 5s SCAN_INTERVAL doesn't make sense to me. If you're so sensitive you better abort and try again rather than waiting for 1.5 minutes.

**Reply by leeyuentuen** [reply]
@ikruglov 
well if it was in 5sec, then there is 6 points (5s, 10s, 15s, 20s, 25s and 30s) where they have check the power.

but normally we should have a timeout on the wallbox unless there is a second connection or the cable is disconnected

Maybe it is the opposite, maybe I'm just the only one that wants this short time, or maybe other people want less fetching

**Comment by leeyuentuen** [comment]
scan interval discussion before:
https://github.com/leeyuentuen/alfen_wallbox/discussions/56

maybe i can add the scan interval in the config form while adding the wallbox. then each person can choose what for interval they want

**Comment by WimBra** [comment]
> @ikruglov, about the SCAN_INTERVAL, at the moment it has been changed to 30sec. don't know if I'm the only one, but I have some automation to change the limit Amps between 6-16A depending on the usage. Because here in Belgium we have 'capacity price' it means the highest Amps that you have to use the latest 15min, it will become the price calculation for the rest of the month. so I'm trying to limit my total usage below 3.5KW, So the reason why I'm trying to put the scan interval as low as possible is to have the possibility to switch to lower amps and even stop the charging if I'm using too much power

You're not the only one :)
I live also in flanders (belgium) with the capacitytarif (15min peak).

pls keep the 5sec interval (like the P1 home-assistant interval) ... or make a variabel input for users for higher interval rate.

**Reply by leeyuentuen** [reply]
plan at the moment is to add a scan interval in the config form when we add the wallbox. so each person can choose self the interval

**Reply by WimBra** [reply]
is the config-form reachable when already installed? or will the standard scan-interval (when updated) be 5s?

**Reply by leeyuentuen** [reply]
i don't think so, but maybe default on 5 sec xD (as it was before that value) for other value, we should remove them and re-add them?

**Comment by ikruglov** [comment]
@leeyuentuen, what is the conclusion? Do you want to implement a selection between options?

**Reply by leeyuentuen** [reply]
i'll later add an option in the config form (where you put the username and password) there will be an extra option to set the scan interval

**Reply by ikruglov** [reply]
LGTM. Thanks

**Reply by leeyuentuen** [reply]
issue create, i'll implement them later

**Reply by leeyuentuen** [reply]
i've create a tmp branch, didn't test them fully yet:
https://github.com/leeyuentuen/alfen_wallbox/tree/scan_interval

**Reply by myhacsint** [reply]
@leeyuentuen I tested it, it seems the interval is still locked at 30sec, I am still seeing only 30 sec. updates - might that be a limitation of the API?

**Reply by leeyuentuen** [reply]
i also notice this. i'll investigate why it doesn't took the scan interval.

**Reply by leeyuentuen** [reply]
@myhacsint the latest beta should solved the problem, can you confirm?

**Reply by myhacsint** [reply]
@leeyuentuen I got 10 sec updates (with a global setting of 5 sec) for the active power, but the power adjustment takes a bit more time.
